6.5.group_by.py

```diff
@@ -0,0 +1,8 @@
+def group_by(func, iterable):
+    mydict = {}
+    for part in iterable:
+        mydict.setdefault(func(part), []).append(part)
+    return mydict
+
+
+print(group_by(len, ['hi', 'bye', 'yo', 'try']))
```

Comment:
Use _name==main_ - 
https://docs.python.org/3/library/__main__.html
https://www.geeksforgeeks.org/python-main-function/amp/

Reply:
fixed it